sdk/src/identifier2.rs

```diff
@@ -0,0 +1,310 @@
+use crate::bytes_serializable::BytesSerializable;
+use crate::error::IggyError;
+use crate::utils::byte_size::IggyByteSize;
+use crate::utils::sizeable::Sizeable;
+use crate::validatable::Validatable;
+use arrayvec::ArrayString;
+use bytes::{BufMut, Bytes, BytesMut};
+use serde::{Deserialize, Serialize};
+use std::borrow::Cow;
+use std::fmt::Display;
+use std::hash::{Hash, Hasher};
+use std::str::FromStr;
+
+/// `Identifier` represents the unique identifier of the resources such as stream, topic, partition, user etc.
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone, Eq)]
+#[allow(clippy::large_enum_variant)]
+pub enum Identifier2 {
+    Numeric(u32),
+    String(ArrayString<255>),
```

Comment:
interestingly this is 260 bytes which brings the whole enum to 264 bytes, we might lower the size to get it right to 256 to help with alignment, wdyt?